mne/io/base.py

```diff
@@ -2500,25 +2500,35 @@ def _write_raw_buffer(fid, buf, cals, fmt):
 
 def _check_raw_compatibility(raw):
     """Ensure all instances of Raw have compatible parameters."""
-    for ri in range(1, len(raw)):
-        if not isinstance(raw[ri], type(raw[0])):
-            raise ValueError(f'raw[{ri}] type must match')
-        for key in ('nchan', 'bads', 'sfreq'):
-            a, b = raw[ri].info[key], raw[0].info[key]
+    for i in range(1, len(raw)):
+        if not isinstance(raw[i], type(raw[0])):
+            raise ValueError(f'raw[{i}] type must match')
+        for key in ('nchan', 'sfreq'):
+            a, b = raw[i].info[key], raw[0].info[key]
             if a != b:
                 raise ValueError(
-                    f'raw[{ri}].info[{key}] must match:\n'
+                    f'raw[{i}].info[{key}] must match:\n'
                     f'{repr(a)} != {repr(b)}')
-        if not set(raw[ri].info['ch_names']) == set(raw[0].info['ch_names']):
-            raise ValueError('raw[%d][\'info\'][\'ch_names\'] must match' % ri)
-        if not all(raw[ri]._cals == raw[0]._cals):
-            raise ValueError('raw[%d]._cals must match' % ri)
-        if len(raw[0].info['projs']) != len(raw[ri].info['projs']):
+        bads0 = set(raw[0].info['bads'])
+        badsi = set(raw[i].info['bads'])
+        bads_mismatch = bads0.symmetric_difference(badsi)
+        if bads_mismatch:
+            raise ValueError(f'raw[{i}][\'info\'][\'bads\'] do not match: '
+                             f'{sorted(bads_mismatch)}')
+        chs0 = set(raw[0].info['ch_names'])
+        chsi = set(raw[i].info['ch_names'])
+        ch_mismatch = chs0.symmetric_difference(chsi)
```

Comment:
This code also could be more DRY -- it should exist in a loop like
```
for kind in ('bads', 'ch_names'):
```
... but this also now does a symmetric diff on the channel names, which I don't think we've discussed yet. I actually don't think this will work -- I'd expect the user to need to do a minual `reorder_channels` before being able to concat successfully/correctly. So for `ch_names` the order does actually matter...

Reply:
@moritz-gerster I think this comment has not been addressed. I do not think you should use `set` with channels like you do (now) for bads, so I think this `chs` part should be reverted / go back to old behavior, ideally with a test that fails if the channel *orders* do not match (it should IIRC!)

Reply:
@moritz-gerster I think this comment has not been addressed. I'll un-resolve it.

I noticed on this read through that it's a `set` check on `main` so you can ignore that part -- but I think it could still be more DRY by using a loop, right?